pubsub/pubsub.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pubsub
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/google/go-cloud/pubsub/driver"
+	"google.golang.org/api/support/bundler"
+)
+
+// Message contains data to be published.
+type Message struct {
+	// Body contains the content of the message.
+	Body []byte
+
+	// Attributes has key/value metadata for the message.
+	Attributes map[string]string
```

Comment:
Metadata

Reply:
I guess Robert is preferring consistency across Go Cloud packages, and I am suggesting consistency across providers. I honestly don't know which is better.

Reply:
I don't feel too strongly about this; I saw that it was renamed `Metadata` in the PR with just the driver and thought that this needed to be consistent.

For blob, many providers have blobs with lots of attributes (Size, ContentType, etc.), and one of the attributes is a set of arbitrary key/value pairs, which they called Metadata.

If pubsub is different, and Messages don't have any attributes other than the key/value pairs, then I'm fine with just calling the key/value pairs attributes.